File: problems/problems_3477/Solution.java
package problems.problems_3477;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public int numOfUnplacedFruits(int[] fruits, int[] baskets) {
        int n = fruits.length;
        int ans = 0;
        for (int fruit: fruits) {
            boolean placed = false;
            for (int i = 0; i < n; i++) {
                if (baskets[i] >= fruit) {
                    baskets[i] = 0;
                    placed = true;
                    break;
                }
            }
            if (!placed) {
                ans++;
            }
        }
        return ans;
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int[] fruits = jsonArrayToIntArray(inputJsonValues[0]);
		int[] baskets = jsonArrayToIntArray(inputJsonValues[1]);
        return JSON.toJSON(numOfUnplacedFruits(fruits, baskets));
    }
}

File: problems/problems_3477/solution.go
package problem3477

import (
	"encoding/json"
	"log"
	"strings"
)

func numOfUnplacedFruits(fruits []int, baskets []int) (ans int) {
	for _, fruit := range fruits {
		placed := false
		for i, basket := range baskets {
			if basket >= fruit {
				baskets[i] = 0
				placed = true
				break
			}
		}
		if !placed {
			ans++
		}
	}
	return
}

func Solve(inputJsonValues string) any {
	inputValues := strings.Split(inputJsonValues, "\n")
	var fruits []int
	var baskets []int

	if err := json.Unmarshal([]byte(inputValues[0]), &fruits); err != nil {
		log.Fatal(err)
	}
	if err := json.Unmarshal([]byte(inputValues[1]), &baskets); err != nil {
		log.Fatal(err)
	}

	return numOfUnplacedFruits(fruits, baskets)
}
